miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -123,6 +125,33 @@ internal class MiniAppDownloader(
         versionId: String
     ) = apiClient.fetchFileList(appId, versionId)
 
+    @Throws(MiniAppSdkException::class)
+    suspend fun fetchMiniAppManifest(appId: String, versionId: String): MiniAppManifest {
+        if (versionId.isEmpty()) throw MiniAppSdkException("Provided Mini App Version ID is invalid.")
+        else {
+            val manifestResponse = apiClient.fetchMiniAppManifest(appId, versionId)
+            return prepareMiniAppManifest(manifestResponse)
+        }
+    }
+
+    @VisibleForTesting
+    fun prepareMiniAppManifest(metadataEntity: MetadataEntity): MiniAppManifest {
+        try {
+            val requiredPermissions = metadataEntity.metadata?.requiredPermissions?.map {
+                Pair(MiniAppCustomPermissionType.getValue(it.name), it.reason)
+            } ?: emptyList()
+
+            val optionalPermissions = metadataEntity.metadata?.optionalPermissions?.map {
+                Pair(MiniAppCustomPermissionType.getValue(it.name), it.reason)
+            } ?: emptyList()
+
+            val customMetadata = metadataEntity.metadata?.customMetaData ?: emptyMap()
+            return MiniAppManifest(requiredPermissions, optionalPermissions, customMetadata)
+        } catch (exception: NullPointerException) {
+            throw MiniAppSdkException("Metadata isn't found in manifest.json for this Mini App.")
```

Comment:
I think you don't need this try/catch anymore, do you? Is there anything that could throw `NullPointerException`?

Reply:
Yeah, now we can remove try/catch. I have updated this part.